sklearn/decomposition/_nmf.py

```diff
@@ -1056,20 +1215,40 @@ def non_negative_factorization(X, W=None, H=None, n_components=None, *,
         if H.dtype != X.dtype:
             raise TypeError("H should have the same dtype as X. Got H.dtype = "
                             "{}.".format(H.dtype))
-        # 'mu' solver should not be initialized by zeros
-        if solver == 'mu':
-            avg = np.sqrt(X.mean() / n_components)
-            W = np.full((n_samples, n_components), avg, dtype=X.dtype)
+
+        if init != 'custom':
+            W, _ = _initialize_nmf(X, n_components, init=init,
+                                   random_state=random_state)
         else:
-            W = np.zeros((n_samples, n_components), dtype=X.dtype)
+            # 'mu' solver should not be initialized by zeros
+            if solver == 'mu':
+                avg = np.sqrt(X.mean() / n_components)
+                W = np.full((n_samples, n_components), avg, dtype=X.dtype)
+            else:
+                W = np.zeros((n_samples, n_components), dtype=X.dtype)
     else:
         W, H = _initialize_nmf(X, n_components, init=init,
                                random_state=random_state)
 
+    if batch_size is not None:
+        if not isinstance(batch_size, numbers.Integral) or batch_size < 0:
+            raise ValueError("Number of samples per batch must be a positive "
+                             "integer; got (batch_size=%r)" % batch_size)
+        if A is None:
+            A = H.copy()
+        if B is None:
+            B = np.ones((n_components, n_features))
+
+        _check_init(A, (n_components, n_features), "NMF (input A)")
+        _check_init(B, (n_components, n_features), "NMF (input B)")
+
     l1_reg_W, l1_reg_H, l2_reg_W, l2_reg_H = _compute_regularization(
         alpha, l1_ratio, regularization)
 
     if solver == 'cd':
+        if batch_size is not None:
+            raise ValueError("Coordinate descent algorithm is not available "
+                             "for MiniBatchNMF. Please set solver to 'mu'.")
```

Comment:
Might want to refer to `batch_size` is not None, instead of the class MiniBatchNMF.